pkg/volume/util/operationexecutor/operations_generator.go

```diff
@@ -0,0 +1,739 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operationexecutor
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	kevents "k8s.io/kubernetes/pkg/kubelet/events"
+	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/util/mount"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+type operationsGenerator struct{}
+
+var _ OperationsGenerator = &operationsGenerator{}
+
+// NewOperationsGenerator returns a new instance of operationExecutorFuncGenerator
+func NewOperationsGenerator() OperationsGenerator {
+	return &operationsGenerator{}
+}
+
+// OperationsGenerator interface that extracts out the functions from operation_executor to make it dependency injectable
+type OperationsGenerator interface {
+	// Generates the MountVolume function needed to perform the mount of a volume
+	GenerateMountVolumeFunc(oe *operationExecutor, waitForAttachTimeout time.Duration, volumeToMount VolumeToMount, actualStateOfWorldMounterUpdater ActualStateOfWorldMounterUpdater) (func() error, error)
```

Comment:
Passing `operationExecutor` in here and then having `OperationGenerator` itself be consumed by `OperationExecutor` is dirty. Instead, figure out what the common dependencies are that these methods need (looks like `volume.VolumePluginMgr`, and others?). Put those dependencies in the constructor (`NewOperationsGenerator(...)`) and save them in your `operationsGenerator struct` for these methods to use. If there are dependencies that are one off for any of these methods, feel free to pass them in as parameters to the method.